mars_street_view/api_call.py

```diff
@@ -0,0 +1,106 @@
+# _*_ Coding: utf-8 _*_
+"""Call the NASA Mars Rover Photo API and return image data."""
+from __future__ import unicode_literals
+
+import os
+import io
+import requests
+import json
+import time
+
+
+SAMPLE_DATA_PATH = os.environ.get('SAMPLE_DATA_PATH')
+
+BASE_URL = 'https://api.nasa.gov/mars-photos/api/v1/rovers/'
+
+ROVERS = {
+    'Curiosity': ''.join((BASE_URL, 'curiosity/photos')),
+    'Opportunity': ''.join((BASE_URL, 'opportunity/photos')),
+    'Spirit': ''.join((BASE_URL, 'spirit/photos')),
+}
+NASA_API_KEY = os.environ.get('NASA_API_KEY')
+
+
+def fetch_photo_data(rover, sol, camera=None):
+    """Make API call to NASA."""
+    try:
+        url = ROVERS[rover]
+    except KeyError:
+        raise ValueError('Incorrect rover name provided.')
+    page = 1
+    lst = []
+    found_ids = set()
+    while True:
+        params = {
+            'sol': sol,
+            'page': page,
+            'api_key': NASA_API_KEY,
+        }
+        if camera:
+            params['camera'] = camera
+        resp = requests.get(url, params=params)
+        if resp.status_code == 400:
+            params['camera'] = camera or ''
+            print('400 response for {0} {camera} sol {sol} page={page}'
+                  ''.format(rover, **params))
+            break
+        content, encoding = resp.content, resp.encoding or 'utf-8'
+        photo_data = json.loads(content.decode(encoding))
+        photos = photo_data.get('photos', [])
+        if not photos:
+            break
+        for photo in photos:
+            if photo['id'] not in found_ids:
+                lst.append(photo)
+                found_ids.add(photo['id'])
+        page += 1
+
+    return lst
+
+
+def fetch_and_save_data_sample():
+    """Download and save json data sample of the first day of each mission."""
+    import pdb; pdb.set_trace()
```

Comment:
not good that there is a breakpoint in your committed code.  I hope this isn't used anywhere.